src/server.rs

```diff
@@ -39,20 +39,26 @@ impl Server {
           None => None,
         };
 
-        let mut client = lnd_client::Client::new(lnd_rpc_authority.clone(), lnd_rpc_cert)
-          .await
-          .context(error::LndGrpcConnect)?;
+        let lnd_rpc_macaroon = match arguments.lnd_rpc_macaroon_path {
+          Some(path) => Some(
+            tokio::fs::read(&path)
+              .await
+              .context(error::FilesystemIo { path })?,
+          ),
+          None => None,
+        };
+
+        let mut client =
+          lnd_client::Client::new(lnd_rpc_authority.clone(), lnd_rpc_cert, lnd_rpc_macaroon)
+            .await
+            .context(error::LndGrpcConnect)?;
 
-        let version = client
-          .get_info()
-          .await
-          .context(error::LndGrpcStatus)?
-          .version;
+        client.ping().await.context(error::LndGrpcStatus)?;
 
         writeln!(
           environment.stderr,
-          "Connected to LND RPC server at {}, version {}",
-          lnd_rpc_authority, version,
+          "Connected to LND RPC server at {}",
+          lnd_rpc_authority
         )
         .context(error::StderrWrite)?;
```

Comment:
I think we should pull out all the lnd related setup from `Server::setup` into its own function.